k8s/apis/federation/v1/catalog_webhook.go

```diff
@@ -78,40 +78,52 @@ func (r *Catalog) validateCreateCatalog() error {
 
 func (r *Catalog) checkSchema() error {
 
-	if schemaName, ok := r.ObjectMeta.Annotations["schema"]; ok {
```

Comment:
Sorry but I forgot about this. Do you think it would be better to have schema reference as part of the ObjectMeta.Annotation or in the Spec.Metadata? If the former, we probably should get rid of Spec.Metadata for good. Sorry for bouncing back and forth, I don't see immediate pros or cons either way, but I'd like to see how you think about it.

Reply:
In function wide, I don't have preference whether ObjectMeta.Annotation is better than Spec.Metadata. 
However considering the default resource mapping in ARM, it won't map the properties in K8S resource's annotation unless you define some liquid templates. So from E2E perspective, you will still schema in properties.metadata in ARM resource together with other spec properties. But in K8S CR, metadata is still in annotation section.
If we keep spec.metadata, the experience will be the same. 
 
I don't think this will affect actual function, only visibility is not the same.

Reply:
let's keep spec.metadata then, make sure the k8s annotation is removed and artifacts are regenerated using code-gen